processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -147,6 +146,23 @@ public void updateSearchResultSet(
     }
   }
 
+  private static boolean isNullSelector(final DimensionSelector selector)
```

Comment:
In this case the parameter should be marked `@Nullable`, but I think we should avoid this, because `null` and "a selector object indistinguishable from `NullDimensionSelector`" are different things. So suggested for `isNullSelector()` to accept only non-null objects, and leave `== null` check outside.

Reply:
Are you suggesting changing the call site to,

```java
if (selector != null && !isNullSelector(selector)) {
  // Do stuff
}
```

Instead of,

```java
if (!isNullSelector(selector)) {
  // Do stuff
}
```


I can make the change, but don't see why one is better than the other in this case… could you please explain your thinking?

Reply:
A simple principle that applies here is that we should minimize nullability of method parameters and return types. It's a source of bugs: selector could be null for a good reason, and could be null by mistake (e. g. uninitialized variable). And even if selector is null for a good reason, accepting it makes the utility method less specific, because there could be variable potential meanings and treatments of selector as null.

Most of this doesn't apply right here, but if this method is used elsewhere and made public, that is very probably, it will apply.

Reply:
Right now this method has exactly one caller, and since it's private in SearchQueryRunner.java it's not going to have more than that unless it's extracted somewhere else. I guess you're saying that someone might want to extract this method to a common utility class in the future, and just in case that happens, we should make the arg non-nullable so the future person is not tempted to create a common utility method with a nullable arg… since in turn that would tempt callers to not handle nulls appropriately.

I guess I can see that point. It seems too theoretical to make it a hard rule in Druid, but I'll do it in this case since we have already spent so much time talking about it. Although in general I think it's fine to be looser with guidelines like this for private helper functions vs. public helper functions, and stricter at the point where someone actually does make it public, if ever.

Reply:
Pushed new helper function.